clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3832,6 +3832,9 @@ def warn_type_attribute_wrong_type : Warning<
   "'%0' only applies to %select{function|pointer|"
   "Objective-C object or block pointer}1 types; type here is %2">,
   InGroup<IgnoredAttributes>;
+def warn_attribute_on_void_param: Warning<
+  "attribute %0 cannot be applied to a 'void' parameter">,
```

Comment:
Is `'void' parameter` technically correct here?  Isn't `(void)` technically an empty parameter list?  Can we word this better, or are we ok with the term?

Reply:
disregard, in the C++ grammar it is a special case in as a parameter.